### ISSUE TITLE
[Bug]: Reproducibility of Gemma-3-270m-it output on A10G gpu

### ISSUE THREAD

**Issue by sujit0892** [vllm-project/vllm#26271]
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>
I am using vllm with temperature=0 and seed 42 for instruction finetuned gemma-3-270m-it. I have followed https://docs.vllm.ai/en/v0.9.1/usage/reproducibility.html to make the llm deterministic still I am getting different results in consecutive run on A10G GPU. However on A100 output is consistent

</details>

### 🐛 Describe the bug

import os
os.environ["VLLM_ENABLE_V1_MULTIPROCESSING"] = "0"
import torch
import random
import numpy as np
seed = 42
torch.manual_seed(seed)
np.random.seed(seed)
random.seed(seed)

sampling_params=SamplingParams(temperature=0,
                                           max_tokens=256,seed=seed)
llm = LLM(model='finetuned-model',dtype='auto',max_model_len=4096)
outputs = llm.generate(prompts, sampling_params)

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
